builder/vsphere/driver/network.go

```diff
@@ -37,7 +37,9 @@ func (n *Network) Info(params ...string) (*mo.Network, error) {
 		p = params
 	}
 	var info mo.Network
-	err := n.network.Properties(n.driver.ctx, n.network.Reference(), p, &info)
+
+	network := n.network.(*object.Network)
```

Comment:
Is there any chance of a panic here?

Reply:
I wasn't 100% sure what to do here. The current implementation only calls Info as part of the builder "test". That test only calls Info after running .NewNetwork(). This should only result in the *object.Network{}. When I looked into the NetworkReference interface, there is not a generic .Properties() function even though they all have it (from Common). The data that you want ends up being passed in (&info in this case). If you have a VirtualSwitch or DVSPortGroup, you would be passing in a different managed object.

Reply:
I'd may be just add a check just for the sake of it:
```suggestion
	network, ok := n.network.(*object.Network)
	if !ok {
		return nil, fmt.Errorf("unexpected %t network object type", n.network)
	}
```

Reply:
One nicer thing you could do here is create a new interface:

```go
type NetworkReference interface {
	object.NetworkReference
	Properties(ctx context.Context, r types.ManagedObjectReference, ps []string, dst interface{}) error
}

type Network struct {
	driver  *Driver
	network NetworkReference
}
```

So you don't have to type-check (cast) anything

Reply:
The interface is going to just move the problem. You need to pass in a specific dst interface depending on what sort of object that you are starting with.